From 3b6dc89febb6b5ca54731dfd43b4d8df2afcb709 Mon Sep 17 00:00:00 2001
From: Yuwei Chen <yuwei.chen@intel.com>
Date: Tue, 8 Sep 2020 14:57:15 +0800
Subject: [PATCH] Support for Add included FakeTest.inc changing - asm file

Description: Support for Add included file FakeTest.inc in the TestPkg/FakeEmuSec/Ia32/SwitchRam.asm.

Repo: edk2

Signed-off-by: Yuwei Chen <yuwei.chen@intel.com>
---
 TestPkg/FakeEmuSec/Ia32/SwitchRam.asm | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/TestPkg/FakeEmuSec/Ia32/SwitchRam.asm b/TestPkg/FakeEmuSec/Ia32/SwitchRam.asm
index f90c569784..140e498d61 100644
--- a/TestPkg/FakeEmuSec/Ia32/SwitchRam.asm
+++ b/TestPkg/FakeEmuSec/Ia32/SwitchRam.asm
@@ -14,7 +14,6 @@
 ;------------------------------------------------------------------------------
 #include "FakeTest.h"
 #include "FakeTest.i"
-#include "FakeTest.inc"
     .586p
     .model  flat,C
     .code
@@ -54,8 +53,8 @@ SecSwitchStack   PROC
     mov   dword ptr [eax], edx
     mov   edx, dword ptr [esp + 4]
     mov   dword ptr [eax + 4], edx
-    mov   edx, dword ptr [esp + testaddr0]
-    mov   dword ptr [eax + testaddr0], edx
+    mov   edx, dword ptr [esp + 6]
+    mov   dword ptr [eax + 6], edx
     mov   edx, dword ptr [esp + testaddr1]
     mov   dword ptr [eax + testaddr1], edx
     mov   edx, dword ptr [esp + testaddr2]    ; Update this function's return address into permanent memory
-- 
2.27.0.windows.1

